custom_components/ics_calendar/calendar.py

```diff
@@ -313,7 +313,9 @@ async def async_get_events(
         event_list: list[ParserEvent] = []
         if await self._calendar_data.download_calendar():
             _LOGGER.debug("%s: Setting calendar content", self.name)
-            self.parser.set_content(self._calendar_data.get())
+            await self._hass.async_add_executor_job(
```

Comment:
So wrapping it inside async_add_executor_job hides the detection?  I wouldn't have expected that.